lib/logstash/pipeline.rb

```diff
@@ -309,4 +319,31 @@ def flush_filters_to_output!(options = {})
     end
   end # flush_filters_to_output!
 
-end # class Pipeline
+  def inflight_count
+    data = {
+      "input_to_filter" => @input_to_filter.size,
+      "filter_to_output" => @filter_to_output.size,
+      "outputs" => []
+    }
+    @outputs.each do |output|
+      next unless output.worker_queue && output.worker_queue.size > 0
+      data["outputs"] << [output.inspect, output.worker_queue.size]
+    end
+
+    data["total"] = data["input_to_filter"] + data["filter_to_output"] +
+                    data["outputs"].map(&:last).inject(0, :+)
+    data
+  end
+
+  def dump
+    dump = []
```

Comment:
this method name shadows the name of the method. Can we change one of the names? (method or variable) ?